examples/exchange-sudt-for-ckb/src/lib.ts

```diff
@@ -0,0 +1,327 @@
+      return outputs.push({
+        cellOutput: {
+          capacity: recycleSUDTCapacity.toHexString(),
+          lock: issuerAccountInfo.lockScript,
+        },
+        data: "0x",
+      });
+    });
+  }
+
+  const inputsCapacity = getTotalCapacity(txSkeleton.inputs);
+  const outputsCapacity = getTotalCapacity(txSkeleton.outputs);
+
+  const CKBChangeCapacity = inputsCapacity.sub(outputsCapacity);
+
+  const CKBChangeOutput: Cell = {
+    cellOutput: {
+      capacity: CKBChangeCapacity.toHexString(),
+      lock: holderAccountInfo.lockScript,
+    },
+    data: "0x",
+  };
+  txSkeleton = txSkeleton.update("outputs", (outputs) => outputs.push(CKBChangeOutput));
+
+  txSkeleton = txSkeleton.update("witnesses", (witnesses) => {
+    const placeholderWitness = {
+      lock:
+        "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+    };
+
+    const witnessArgs = bytes.hexify(blockchain.WitnessArgs.pack(placeholderWitness));
+
+    for (let index = 0; index < txSkeleton.inputs.size; index++) {
+      witnesses = witnesses.push(witnessArgs);
+    }
+
+    return witnesses;
+  });
+
+  txSkeleton = await payFeeByFeeRate(txSkeleton, [holderAccountInfo.address], 1000, undefined, { config: AGGRON4 });
+
+  // STEP2: sign the transaction skeleton
+  const tx = signTx(txSkeleton, [issuerPrivateKey, holderPrivateKey, issuerPrivateKey]);
+
+  // STEP3: send it to testnet
+  const txHash = await rpc.sendTransaction(tx, "passthrough");
+  return txHash;
+}
+
+/**
+ * sign a transaction
+ */
+export function signTx(txSkeleton: helpers.TransactionSkeletonType, privateKey: string[]) {
+  txSkeleton = commons.common.prepareSigningEntries(txSkeleton);
+  const signatures = txSkeleton.get("signingEntries").map(({ message }, index) => {
+    return hd.key.signRecoverable(message, privateKey[index]);
+  });
+  const tx = helpers.sealTransaction(txSkeleton, signatures.toJSON());
+  return tx;
+}
+
+export function calculateSUDTAmountSum(cells: Cell[]) {
+  let amount = BI.from(0);
+  for (const cell of cells) {
+    if (cell.cellOutput.type?.codeHash === AGGRON4.SCRIPTS.SUDT.CODE_HASH) {
+      amount = amount.add(number.Uint128LE.unpack(cell.data));
+    }
+  }
+
+  return amount;
+}
+
+export function calculateCKBSum(cells: Cell[]) {
+  let amount = BI.from(0);
+
+  for (const cell of cells) {
+    amount = amount.add(cell.cellOutput.capacity);
+  }
+
+  return amount;
+}
+
+/**
+ * collect all SUDT cells and calculate their data sum
+ * @param address SUDT Issuer address
+ * @returns the amount of SUDT in the address
+ */
+export async function fetchSUDTBalance(address: string) {
+  const collector = indexer.collector({
+    lock: helpers.parseAddress(address, { config: AGGRON4 }),
+    type: {
+      script: {
+        codeHash: AGGRON4.SCRIPTS.SUDT.CODE_HASH,
+        hashType: AGGRON4.SCRIPTS.SUDT.HASH_TYPE,
+        args: "0x",
```

Comment:
why leave 0x for args here, I think this should be some specific args

Reply:
Yes. `0x` is not the best here. Although we only have 1 type SUDT in example. but a specific args prop is more correct. And more meaningful for teaching users.